sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,264 @@
+  - [Other](#other)
+- [Additional information](#additional-information)
+
+## Migration benefits
+
+To understand why we created our version 12 client libraries, you may refer to the Tech Community blog post, [Announcing the Azure Storage v12 Client Libraries](https://techcommunity.microsoft.com/t5/azure-storage/announcing-the-azure-storage-v12-client-libraries/ba-p/1482394) or refer to our video [Introducing the New Azure SDKs](https://aka.ms/azsdk/intro).
+
+Included are the following:
+- Thread-safe synchronous and asynchronous APIs
+- Improved performance
+- Consistent and idiomatic code organization, naming, and API structure, aligned with a set of common guidelines
+- The learning curve associated with the libraries was reduced
+
+Note: The blog post linked above announces deprecation for previous versions of the library.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespaces root for version 12 Azure client libraries follow the pattern `Azure.[Area].[Service]` where the legacy libraries followed the pattern `Microsoft.Azure.[Area].[Service]`.
+
+In this case, to install the legacy v11 package with Nuget:
+```
+dotnet add package Microsoft.Azure.Storage.Blob
+```
+
+It is now the following for v12:
+```
+dotnet add package Azure.Storage.Blobs
+```
+
+### Authentication
+
+#### Managed Identity
+
+v11
+
+TODO
+
+v12
+
+TODO
+
+#### SAS
+
+There are various SAS tokens that may be generated. Visit our documentation pages to learn how to [Create a User Delegation SAS](https://docs.microsoft.com/azure/storage/blobs/storage-blob-user-delegation-sas-create-dotnet), [Create a Service SAS](https://docs.microsoft.com/azure/storage/blobs/storage-blob-service-sas-create-dotnet), or [Create an Account SAS](https://docs.microsoft.com/azure/storage/common/storage-account-sas-create-dotnet?toc=/azure/storage/blobs/toc.json).
+
+v11
+```csharp
+TODO
+```
+
+v12
+```csharp
+// This code snippet creates a service level SAS that only allows reading
+// from service level APIs
+AccountSasBuilder sas = new AccountSasBuilder
+{
+    // Allow access to blobs
+    Services = AccountSasServices.Blobs,
+    // Allow access to the service level APIs
+    ResourceTypes = AccountSasResourceTypes.Service,
+    // Access expires in 1 hour!
+    ExpiresOn = DateTimeOffset.UtcNow.AddHours(1)
+};
+// Allow read access
+sas.SetPermissions(AccountSasPermissions.Read);
+// Create a SharedKeyCredential that we can use to sign the SAS token
+StorageSharedKeyCredential credential = new StorageSharedKeyCredential(StorageAccountName, StorageAccountKey);
+// Build a SAS URI
+UriBuilder sasUri = new UriBuilder(StorageAccountBlobUri);
+sasUri.Query = sas.ToSasQueryParameters(credential).ToString();
+// Create a client that can authenticate with the SAS URI
+BlobServiceClient service = new BlobServiceClient(sasUri.Uri);
+// Make a service request to verify we've successfully authenticated
+await service.GetPropertiesAsync();
+```
+
+Summary:
```

Comment:
What is this meant to be?

Reply:
It was meant to be a brief description of the underlying changes we made to give the developer a better idea of the differences they're looking at. But it could be unnecessary too. If the summaries don't add much value then it doesn't make sense to keep them. What do you think?